frontend/src/react/Components/generic/SelectQuantity/SelectQuantity.jsx

```diff
@@ -7,21 +7,29 @@ import classNames from './selectquantity.module.css';
 
 export default class SelectQuantity extends React.PureComponent {
     colCounts = {};
+    studcolCounts = {};
     constructor() {
         super();
         this.state = {
             colCountKey: 2,
+            studcolCountKey: 2,
         };
 
-        [ 1, 2, 3, 4, 5, 6, 7, 8].forEach((value, i) => { this.colCounts[i] = value; });
+        [1, 2, 3, 4, 5, 6, 7, 8].forEach((value, i) => { this.colCounts[i] = value; });
+        [1,2, 3, 4, 5, 6, 7, 8, 9, 10].forEach((value, i) => { this.studcolCounts[i] = value; });
+    }
+    onStudColCountChange = (studcolCountKey) => {
+        this.props.onStudColCountChange(studcolCountKey);
+        this.setState({ studcolCountKey });
     }
     onColCountChange = (colCountKey) => {
-        this.props.onColCountChange(colCountKey)
+        this.props.onColCountChange(colCountKey);
         this.setState({ colCountKey });
```

Comment:
ça fonctionne mais ton setState n'est pas écrit comme on le fait ailleurs.

déclare tes states comme ça : state = {colCounts : [valeur initiale],...}

Reply:
Ensuite dans ton setState fait plutôt : this.setState({colCounts : colCounts}) pour qu'on sache bien de quoi tu parles

Reply:
@CyrilBjn peut tu voir ca, et les indentations ?